lib/automate.js

```diff
@@ -0,0 +1,120 @@
+var BaseClient = require("./client");
+var extend = require("./extend");
+var util = require("util");
+
+function AutomateClient(settings) {
+  this.server = {
+    host: "www.browserstack.com"
+  };
+  BaseClient.call(this, settings);
+}
+
+util.inherits(AutomateClient, BaseClient);
+
+// public API
+extend(AutomateClient.prototype, {
+
+  getPlan: function(fn) {
+    this.request({
+      path: this.path("/plan.json")
+    }, fn);
+  },
+
+  getBrowsers: function(fn) {
+    this.request({
+      path: this.path("/browsers.json")
+    }, fn);
+  },
+
+  getProjects: function(fn) {
+    this.requestAutomate({
+      path: this.path("/projects.json")
+    }, fn);
+  },
+
+  getProject: function(id, fn) {
+    this.requestAutomate({
+      path: this.path("/projects/" + id + ".json")
+    }, fn);
+  },
+
+  getBuilds: function(options, fn) {
+    if (typeof fn === "undefined") {
+      fn = options;
+      options = {};
+    }
+    this.requestAutomate({
+      path: this.path("/builds.json")
+    }, fn);
+  },
+
+  getSessions: function(buildId, options, fn) {
+    if (typeof fn === "undefined") {
+      fn = options;
+      options = {};
+    }
+    this.requestAutomate({
```

Comment:
Add a blank line above this to separate the parameter normalization from the action.